pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -61,24 +61,27 @@ func getEnvTemplate(cr *v1.KieApp) v1.EnvTemplate {
 	if cr.Spec.KieDeployments == 0 {
 		cr.Spec.KieDeployments = 1
 	}
-	if len(cr.Spec.Objects.Console.Env) == 0 {
-		cr.Spec.Objects.Console.Env = []corev1.EnvVar{{Name: "empty"}}
-	}
-	if len(cr.Spec.Objects.Server.Env) == 0 {
-		cr.Spec.Objects.Server.Env = []corev1.EnvVar{{Name: "empty"}}
-	}
 	if cr.Spec.RhpamRegistry == (v1.KieAppRegistry{}) {
 		cr.Spec.RhpamRegistry.Registry = logs.GetEnv("REGISTRY", constants.RhpamRegistry) // default to red hat registry
 		cr.Spec.RhpamRegistry.Insecure = logs.GetBoolEnv("INSECURE")
 	}
 
+	consoleName := constants.RhpamcentrServicePrefix
+	consoleImage := constants.RhpamcentrImageName
+	if cr.Spec.KieConsoleType == v1.KieConsoleTypeMonitoring {
+		consoleName = constants.RhpamcentrMonitoringServicePrefix
+		consoleImage = constants.RhpamcentrMonitoringImageName
+	}
```

Comment:
The purpose of the consoleName + consoleImage constants is to be able to reuse the deploymentConfig defined in common.yaml for both Authoring and Monitoring. Otherwise we will need to duplicate all the resources for businesscentral-monitoring.
By adding the consoleType we will be able to decide which one to deploy. The only place to decide the type before the template processing is the CR and that's why I decided to add the `consoleType` parameter.

Reply:
Another way of doing this without having the `consoleType` can be based on the environment name (i.e. `prod`).
However, we can also duplicate all the resources for bc-monitoring and maintain them independently but I'd rather avoid this option given the similarities of both deployments.

Reply:
I think the more important principle is the idea of providing prescriptive environments. This would take something away from that by having the CR determine the type of workbench, in addition to or regardless of the specified environment name. Maybe this can be solved by allowing consoleType to be specified in the environment file itself?

Another caveat is that while the only difference between full workbench and monitoring is the image today, that may soon change. Evan's PR removes the need for controller persistence, at which point the monitoring DC would not longer need a PVC/PV.

Reply:
The consoleType can't be set in the env file because the env file is parsed all at once as a template where the variables should be known beforehand.
Regarding the differences, if the PV/PVC is the only difference, the rest of the deployment remains the same. That leaves the image name, and the resources' names, if we wan't to stick to the templates deployment (which might not be a must) being the differences.
So I still believe we should be able to have a common starting point for both consoles. How the console type is decided can be based upon the name of the env as I suggested before.
WDYT?

Reply:
If/when removing the PVC for monitoring, what do you do with the [volume reference](https://github.com/kiegroup/kie-cloud-operator/blob/master/config/envs/production.yaml#L16-L19) in the DC?

If I understand correctly, you mean keeping {{.ConsoleName}} and {{.ConsoleImage}}, but instead of using values from the CR to populate them, to use the environment name? That should work, though it might be good to think about the pros and cons first. It would allow us to avoid duplication of the console DC. One downside I can think of is how much work customers would have to do to create their own environments (update the configmap) to flip the name/image. I suppose they can also just edit the common env for that. Anything else you can think of?

Unrelatedly, we want to "release" this for the Cloud team to work on after this week. I think we're far behind on some more urgent updates, including the missing environment (Tommy is on PTO and can't remember if he started on that at all) and at least an example of configuration updates through the CR (e.g., new SSO, or ldap entries in CRD).

Reply:
The current volume references will be then moved to the individual envs (trial/authoring)

Regarding the new envs... it should be just adding the name of the new env to the MonitoringEnvs map

Finally, I'll work on a different PR for the missing envs.